evaluation/what-to-support.md

```diff
@@ -0,0 +1,152 @@
+# What should we support
+
+There are a lot of things we would ideally support, this file is intended as a collection of all of them. Not all of them will end up being supported as they may conflict with other design and implementation goals.
+
+## purely const
+
+Dealing with `const`-ness does not change type signatures.
+
+```rust
+impl<const N: usize, T> [T; N] {
+    // `map` is const if `F` is const.
+    //
+    // This is the case as `map` eagerly uses `f` in
+    // its function body.
+    const<C> fn map<U, F, const<C>>(self, f: F) -> [U; N]
+    where
+        F: const<C> FnMut(T) -> U,
+    {
+        // ...
+    }
+}
+
+trait Iterator {
+    type Item;
+    // `map` is always `const`.
```

Comment:
**is** is wrong, as the `Iterator` trait is already stable, so we can only change this to `const<C>` map for `const<C> Iterator`

Reply:
Are you saying that the `const<C>` param must necessarily live on the trait, rather than the method?

Reply:
yes, `fn map` must be conditionally `const` to be backwards compatible